lib/timers.js

```diff
@@ -481,6 +481,7 @@ Timeout.prototype.unref = function() {
     }
 
     var handle = reuse(this);
+    delete handle._list;
```

Comment:
Why not removing the reference to the `TimersList` instance in `reuse`'s implementation instead? 

Also, #8372 hasn't landed yet, and it's not clear whether it will land. As a result I would think that if deleting the `._list` property is more costly than setting it to `null` (or doing anything else that we know might perform better), we should not do that and instead do what we know performs better.